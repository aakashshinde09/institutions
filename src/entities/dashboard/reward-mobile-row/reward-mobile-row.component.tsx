import type { RewardLookup } from '@api/data-contracts'
import { useGetNgoListQuery } from '@shared/api'
import { formatDate, formatNumber } from '@shared/lib/utils'
import { CaptionItem } from '@shared/ui/caption-item'
import { useMemo } from 'react'

interface IRewardMobileRowProperties {
  data: RewardLookup
  index: number
}

export const RewardMobileRow = ({ data, index }: IRewardMobileRowProperties) => {
  const { data: allNgos } = useGetNgoListQuery()
    
  const currentNgo = useMemo(() => {
    const ngo = allNgos?.data?.items?.find((x) => x.ngoAddress === data.ngoAddress)
    return ngo ?? { name: 'Unknown', imageLink: '' }
  }, [allNgos?.data?.items, data.ngoAddress])

  return (
    <div className="rounded-3xl bg-white p-4">
      <div className="flex items-center justify-between">
        <img
          src={currentNgo.imageLink}
          alt={currentNgo.name}
          className="h-6 w-6 rounded-full"
        />
        <div className="text-sm text-muted">{formatDate(data.createdAt)}</div>
      </div>
      <div className="mt-2 flex items-center justify-between">
        <h4 className="font-bold">{currentNgo.name}</h4>
        <div className="font-semibold">#{index}</div>
      </div>
      <div className="mt-6 grid grid-cols-2 gap-y-4">
        <CaptionItem
          title="Staked amount"
          value={`${formatNumber(data.balanceWithoutProfit)} stETH`}
        />
        <CaptionItem
          title="Your profit"
          value={`${formatNumber(data.profit, true)} stETH`}
        />
        <CaptionItem title="Donation" value={`${formatNumber(data.donation, true)} stETH`} />
        <CaptionItem
          title="Service fee"
          value={`${formatNumber(data.serviceFee, true)} stETH`}
        />
      </div>
    </div>
  )
}
