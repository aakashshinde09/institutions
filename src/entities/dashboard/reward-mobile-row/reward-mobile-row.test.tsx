import { expect, test, describe } from 'vitest'
import { render } from '@testing-library/react'
import { RewardMobileRow } from './reward-mobile-row.component'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'

const mock = {
  balanceWithoutProfit: 0.01,
  createdAt: '2024-05-05T19:28:02.000Z',
  donation: 0.000000049,
  id: 241,
  ngoAddress: '0xa72afa79826df92b88831b9ebd1a459a75345231',
  profit: 0.00000019,
  serviceFee: 1.299552152886571e-9,
  stackedAmount: 0.01,
}

describe('<RewardMobileRow/>', () => {
  test('Renders correctly', () => {
    const component = render(
      <QueryClientProvider client={new QueryClient()}>
        <RewardMobileRow data={mock} index={1} />
      </QueryClientProvider>,
    )

    expect(component).toMatchSnapshot()
  })
})
