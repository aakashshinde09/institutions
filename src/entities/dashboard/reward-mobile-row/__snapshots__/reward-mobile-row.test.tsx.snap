// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<RewardMobileRow/> > Renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rounded-3xl bg-white p-4"
      >
        <div
          class="flex items-center justify-between"
        >
          <img
            alt="Unknown"
            class="h-6 w-6 rounded-full"
            src=""
          />
          <div
            class="text-sm text-muted"
          >
            05/05/2024
          </div>
        </div>
        <div
          class="mt-2 flex items-center justify-between"
        >
          <h4
            class="font-bold"
          >
            Unknown
          </h4>
          <div
            class="font-semibold"
          >
            #
            1
          </div>
        </div>
        <div
          class="mt-6 grid grid-cols-2 gap-y-4"
        >
          <div
            class="truncate"
          >
            <p
              class="font-semibold max-md:text-sm max-md:font-normal"
            >
              Staked amount
            </p>
            <span
              class="mt-2 text-2xl font-bold max-md:text-base"
            >
              0.01 stETH
            </span>
          </div>
          <div
            class="truncate"
          >
            <p
              class="font-semibold max-md:text-sm max-md:font-normal"
            >
              Your profit
            </p>
            <span
              class="mt-2 text-2xl font-bold max-md:text-base"
            >
              0.00000019 stETH
            </span>
          </div>
          <div
            class="truncate"
          >
            <p
              class="font-semibold max-md:text-sm max-md:font-normal"
            >
              Donation
            </p>
            <span
              class="mt-2 text-2xl font-bold max-md:text-base"
            >
              0.000000049 stETH
            </span>
          </div>
          <div
            class="truncate"
          >
            <p
              class="font-semibold max-md:text-sm max-md:font-normal"
            >
              Service fee
            </p>
            <span
              class="mt-2 text-2xl font-bold max-md:text-base"
            >
              0.0000000013 stETH
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rounded-3xl bg-white p-4"
    >
      <div
        class="flex items-center justify-between"
      >
        <img
          alt="Unknown"
          class="h-6 w-6 rounded-full"
          src=""
        />
        <div
          class="text-sm text-muted"
        >
          05/05/2024
        </div>
      </div>
      <div
        class="mt-2 flex items-center justify-between"
      >
        <h4
          class="font-bold"
        >
          Unknown
        </h4>
        <div
          class="font-semibold"
        >
          #
          1
        </div>
      </div>
      <div
        class="mt-6 grid grid-cols-2 gap-y-4"
      >
        <div
          class="truncate"
        >
          <p
            class="font-semibold max-md:text-sm max-md:font-normal"
          >
            Staked amount
          </p>
          <span
            class="mt-2 text-2xl font-bold max-md:text-base"
          >
            0.01 stETH
          </span>
        </div>
        <div
          class="truncate"
        >
          <p
            class="font-semibold max-md:text-sm max-md:font-normal"
          >
            Your profit
          </p>
          <span
            class="mt-2 text-2xl font-bold max-md:text-base"
          >
            0.00000019 stETH
          </span>
        </div>
        <div
          class="truncate"
        >
          <p
            class="font-semibold max-md:text-sm max-md:font-normal"
          >
            Donation
          </p>
          <span
            class="mt-2 text-2xl font-bold max-md:text-base"
          >
            0.000000049 stETH
          </span>
        </div>
        <div
          class="truncate"
        >
          <p
            class="font-semibold max-md:text-sm max-md:font-normal"
          >
            Service fee
          </p>
          <span
            class="mt-2 text-2xl font-bold max-md:text-base"
          >
            0.0000000013 stETH
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
